### ISSUE TITLE
Strategy for committing build artifacts

### ISSUE THREAD

**Issue by dmethvin-gov** [usds/us-forms-system#218]
For a fleeting moment we had a git pre-push hook in place but [it didn't do what we wanted](https://github.com/usds/us-forms-system/pull/215#issuecomment-412091307) anyway. 

If we are going to commit build artifacts with every commit, we need some way to ensure that they exactly represent the source committed along with them. It's a bit tricky, the CI can't do it because it's too late by then. We need to have each PR rebased against master at the point where it's landed and *then* rebuild the artifacts before committing. I don't see any [GitHub hooks or API](https://github.com/github/platform-samples) to do that. It's doable if one of us lands the PRs manually.

One possibility is to *not* commit built artifacts except on `production`. That wouldn't affect users of the library since they should be using versioned releases that would still include build artifacts. There are cons to that since it makes testing PRs a two-step process (you have to build after pulling).

**Comment by jcmeloni-usds** [comment]
@dmethvin-gov Did we land on a strategy/plan? I've never been a fan of committing any build artifacts, but don't know the backstory here if there's a reason that we started doing it and it needs to stay.

**Comment by dmethvin-gov** [comment]
I completely agree, I'd prefer to not commit any. Right now vets.gov is pulling in builds via a GitHub hash so they needed it in the past, but I think it's safe to get rid of them now. The npm releases will have everything built.

After our discussions with the vets.gov team yesterday I have some thoughts about how to do this, once we've finished Phase 1 I can get this done.
